fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -134,112 +137,181 @@ public static void listBuckets(AmazonS3 cosClient) {
         }
     }
 
-    private static int getFhirResourceFromBufferReader(BufferedReader resReader, int numOfLinesToSkip, List<Resource> fhirResources) throws Exception {
+    /**
+     * @param resReader - the buffer reader to read FHIR resource from.
+     * @param numOfProcessedLines - number of the already processed lines.
+     * @param fhirResources - List holds the FHIR resources.
+     * @param isSkipProcessed - if need to skip the processed lines before read.
+     * @return - the number of parsing failures.
+     * @throws Exception
+     */
+    private static int getFhirResourceFromBufferReader(BufferedReader resReader, int numOfProcessedLines, List<Resource> fhirResources,
+            boolean isSkipProcessed, String dataSource) throws Exception {
         int exported = 0;
         int lineRed = 0;
+        int parseFailures = 0;
+
         String resLine = null;
         do {
             resLine = resReader.readLine();
             if (resLine != null) {
                 lineRed++;
-                if (lineRed <= numOfLinesToSkip) {
+                if (isSkipProcessed && lineRed <= numOfProcessedLines) {
                     continue;
                 }
-                fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
-                exported++;
-                if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
-                    break;
+                try {
+                    fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
+                    exported++;
+                    if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
+                        break;
+                    }
+                } catch (FHIRParserException e) {
+                    // Log and skip the invalid FHIR resource.
+                    logger.warning("getFhirResourceFromBufferReader: " + e.getMessage());
```

Comment:
why not log the stack trace?  might it have some useful info for figuring out what is invalid about the resource that failed to parse?

Reply:
I thought stack track is a little bit too heavy, with exception and the line number should be good enough for following up if need...

Reply:
I thought stack trace is a little bit too heavy in the log, with the exception info and the line number, we should already have enough info to follow up if need ...

Reply:
OK.  If not taking my suggestion, I'd still suggest to log it at level INFO.   This will be more consistent with the way we handle invalid data sent by the client on the REST side.

Reply:
ok, sounds good

Reply:
accepted you change and fixed the error.